cluster-ui: add index recs to v2 table details page

xinhaoz opened the pull request:
This commit populates the index view in the v2 table details page.  It also replaces usages of
FormatWithTimezone to <Timestamp /> where appropriate.

Closes: #130687
Epic: CRDB-37558
Release note: None

xinhaoz committed:
cluster-ui: add index recs to v2 table details page

This commit populates the index view in the v2 table
details page.  It also replaces usages of
FormatWithTimezone to <Timestamp /> where appropriate.

Closes: #130687
Epic: CRDB-37558
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/131908)
<!-- Reviewable:end -->

xinhaoz committed:
cluster-ui: rename ColumnTitle Tooltip

Reuse this component as a more generic tooltip component
with an underline, which is commonly applied through DB
console.

Epic: none

Release note: None

xinhaoz commented:
https://www.loom.com/share/d7f9d98938924b59946f05f5b57d312a

kyle-a-wong commented on pkg/ui/workspaces/cluster-ui/src/databaseDetailsV2/tablesView.tsx:
```diff
@@ -100,12 +92,12 @@ const COLUMNS: (TableColumnProps<TableRow> & { sortKey?: TableSortOption })[] =
-          }}
-        />
+        <Tooltip
+          underline
```
Is this prop doing anything? I see that `ToolTip` component only accesses `props.noUnderline,` `props.children` and `props.title`

xinhaoz replied:
Oops, this was leftover from a previous change. Removed.

kyle-a-wong commented on pkg/ui/workspaces/cluster-ui/src/indexDetailsPage/indexDetailsPage.tsx:
```diff
@@ -384,15 +383,10 @@ export class IndexDetailsPage extends React.Component<
     return (
       <Breadcrumbs
```
Im assuming that the database tables breadcrumb was removed because there is no database Id to use for the link. Should we fetch it via `/api/v2/table_metadata/<details.table_id>`? 

I think this would be useful to make the breadcrumb consistent with the other pages (will also allow for us to use the same table name as other pages, instead of the of the quoted table names like in the legacy pages

xinhaoz replied:
It's a bit awkward to introduce this at the moment due to this being a legacy page. We could simplify it if we don't use SWR or just use the fetch call directly but I'm not sure if it's worth it. Alternative is to add the dbId to the index stats response. Thoughts? We can do that in a follow-up.

xinhaoz commented:
@kyle-a-wong  Filed #131908.

xinhaoz commented:
tftr! PR incoming for issue above.
bors r+
